### PR TITLE
[#607] Added .internal_name field to Form and FormDefinition

### PR DESCRIPTION
**Pull request by Bartvaderkin**

### CODE REVIEW

File: src/openforms/js/components/admin/form_design/FormMetaFields.js [hunk 1]
```diff
@@ -78,6 +79,16 @@ const FormMetaFields = ({
...
                 </Field>
             </FormRow>
+            <FormRow>
+                <Field
+                    name="form.internalName"
+                    label={<FormattedMessage defaultMessage="Internal name" description="Form name field label" />}
+                    helpText={<FormattedMessage defaultMessage="Internal name/title of the form" description="Form name field help text" />}
+                    errors={errors.name}
+                >
+                    <TextInput value={internalName} onChange={onChange} onBlur={setFormSlug} />
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/FormMetaFields.js:89]:
```suggestion
                    <TextInput value={internalName} onChange={onChange} />
```
Slug should be generated from public name, as it's used in URLs

File: src/openforms/js/components/admin/form_design/FormStepDefinition.js [hunk 2]
```diff
@@ -73,6 +73,17 @@ const FormStepDefinition = ({ url='', name='', slug='', previousText='', saveTex
...
+                    <Field
+                        name="internalName"
+                        label={<FormattedMessage defaultMessage="Internal step name" description="Form step internal name label" />}
+                        helpText={<FormattedMessage
+                            defaultMessage="Internal name of the form definition used in this form step"
+                            description="Form step internal name field help text" />}
+                        errors={errors.name}
+                        fieldBox
+                    >
+                        <TextInput value={internalName} onChange={onFieldChange} onBlur={setSlug} />
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/form_design/FormStepDefinition.js:85]:
```suggestion
                        <TextInput value={internalName} onChange={onFieldChange} />
```
Slug should be set based on public name

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Minor remark about the slugify functionality, rest is ok

**Comment by Bartvaderkin** [comment]
Ugh, copy paste mistake. Should be fixed now.

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**